From bb1feb52ef4368469ecb9f62ff76b481bb3b03b2 Mon Sep 17 00:00:00 2001
From: Florian Westphal <fwestpha@redhat.com>
Date: Thu, 4 Sep 2014 11:21:53 -0400
Subject: [net] netfilter: conntrack: Fix UP builds

Message-id: <1409829716-13720-7-git-send-email-fwestpha@redhat.com>
Patchwork-id: 91224
O-Subject: [PATCH 6/9 RHEL7.1 net] netfilter: conntrack: Fix UP builds
Bugzilla: 1043012
RH-Acked-by: Daniel Borkmann <dborkman@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Jesper Brouer <jbrouer@redhat.com>

BZ: 1043012

commit d5d20912d33f13766902a27087323f5c94e831c8
Author: Eric Dumazet <edumazet@google.com>
Date:   Mon Mar 17 13:37:53 2014 -0700

    netfilter: conntrack: Fix UP builds

    ARRAY_SIZE(nf_conntrack_locks) is undefined if spinlock_t is an
    empty structure. Replace it by CONNTRACK_LOCKS

    Fixes: 93bb0ceb75be ("netfilter: conntrack: remove central spinlock nf_conntrack_lock")
    Reported-by: kbuild test robot <fengguang.wu@intel.com>
    Signed-off-by: Eric Dumazet <edumazet@google.com>
    Cc: Jesper Dangaard Brouer <brouer@redhat.com>
    Cc: Pablo Neira Ayuso <pablo@netfilter.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/nf_conntrack_core.c b/net/netfilter/nf_conntrack_core.c
index 40f596c..aa5dd62 100644
--- a/net/netfilter/nf_conntrack_core.c
+++ b/net/netfilter/nf_conntrack_core.c
@@ -1685,7 +1685,7 @@ int nf_conntrack_init_start(void)
  int max_factor = 8;
  int i, ret, cpu;
 
- for (i = 0; i < ARRAY_SIZE(nf_conntrack_locks); i++)
+ for (i = 0; i < CONNTRACK_LOCKS; i++)
   spin_lock_init(&nf_conntrack_locks[i]);
 
  /* Idea from tcp.c: use 1/16384 of memory.  On i386: 32MB
-- 
1.7.1